payu/subcommands/run_cmd.py

```diff
@@ -147,6 +147,7 @@ def runscript():
             expt.manifest = Manifest(expt.config.get('manifest', {}),
                                      reproduce=False)
             expt.set_output_paths()
+            expt.set_environment_vars()
```

Comment:
Just a small idea: wondering whether it would work if the new function was called within `run_userscript` from `experiment.py`, e.g:

```python
def run_userscript(self, script_cmd):
        self.set_environment_vars()

        # First try to interpret the argument as a full command:
        try:
            sp.check_call(shlex.split(script_cmd))
```
I was thinking this could make it so that the variables are only exported in situations when they might be needed and it might make the purpose of the new function a bit clearer when skimming through the code.

Reply:
Yeah that makes more sense for  it to be run there and ensures the env vars have been set before any user-script is run. I've now added that in